libkbfs/file_data.go

```diff
@@ -901,15 +953,46 @@ func (fd *fileData) shiftBlocksToFillHole(
 			immedParent.pblock.IPtrs[currIndex],
 			newImmedParent.pblock.IPtrs[newCurrIndex]
 
-		// Cache the new immediate parent as dirty.
+		// Cache the new immediate parent as dirty.  Also cache the
+		// old immediate parent's right-most leaf child as dirty, to
+		// make sure this path is captured in
+		// getNextDirtyFileBlockAtOffset calls.  TODO: this is
```

Comment:
Since you only need the previous immediate parent, why do you need to cache the leaf? Why not just cache the old immediate parent as dirty?